src/rust/src/lazy/dsl.rs

```diff
@@ -736,52 +733,20 @@ impl Expr {
 
     //Note clip is implemented a bit different that py-polars
     //instead of PyValue -> AnyValue , it goes Robj -> Literal Expression -> AnyValue
-    pub fn clip(&self, min: &Expr, max: &Expr) -> List {
-        use crate::rdatatype::literal_to_any_value;
-        let expr_res = || -> Result<Expr, String> {
-            match (min.clone().0, max.clone().0) {
-                (pl::Expr::Literal(mi), pl::Expr::Literal(ma)) => {
-                    let av_min = literal_to_any_value(mi)?;
-                    let av_max = literal_to_any_value(ma)?;
-                    Ok(Expr(self.0.clone().clip(av_min, av_max)))
-                }
-                (mi, pl::Expr::Literal(_)) => Err(format!("min [{:?}] was not a literal:", mi)),
-                (pl::Expr::Literal(_), ma) => Err(format!("max [{:?}] was not a literal:", ma)),
-                (mi, ma) => Err(format!(
-                    "neither min [{:?}] or max[{:?}] were literals:",
-                    mi, ma
-                )),
-            }
-        }();
-        r_result_list(expr_res)
+    pub fn clip(&self, min: Robj, max: Robj) -> RResult<Self> {
+        let av_min = robj_to!(PLExpr, min)?;
+        let av_max = robj_to!(PLExpr, max)?;
+        Ok(Expr(self.0.clone().clip(av_min, av_max)))
     }
```

Comment:
Not sure about this, but at least it compiles. Maybe it will need to be fixed when we look at the tests on R side